### ISSUE TITLE
Error: "invalid database read for database" for some, but not all, files.

### ISSUE THREAD

**Issue by gmteunisse** [soedinglab/MMseqs2#35]
I'm trying to create clusters of a number of pre-clustered databases, all stored in multi-fasta files. The `createdb` command seems to runs fine on all files, however, the `cluster` command leads to an error on some, but not all, files:

```
Invalid database read for database data file=tmp_clusters/DB, database index=tmp_clusters/DB.index
getData: local id (27) >= db size (27)
Rescore diagonals.
Could not open data file tmp_clusters/tmp/linclust/pref!
Could not open data file tmp_clusters/tmp/linclust/pref_rescore1!
awk: can't open file tmp_clusters/tmp/linclust/pre_clust.index
 source line number 1
```

This error is followed by many other errors, all related to being unable to open data files (e.g. `Could not open data file tmp_clusters/tmp/linclust/pref!`).

The error seems to depend on the sequences in the file. For example, if I merge two of the fasta-files, one of which runs without errors and one which leads to above error, the error is reproduced. I have not been able to identify which feature in the sequences leads to the error. Before every run of mmseqs, I delete all temp files.

Regarding my environment:
* OSX 10.11.3
* Git commit - 3ff5b8e26c93420c04c589fc663734ee8c964c3c
* Self compiled 

I've attached two files, one that leads to errors in my build, and one that does not. Hope you can help. 

[DB_errors.txt](https://github.com/soedinglab/MMseqs2/files/1089709/DB_errors.txt)
[DB_no_errors.txt](https://github.com/soedinglab/MMseqs2/files/1089710/DB_no_errors.txt)

**Comment by martin-steinegger** [comment]
This error is linked to the amount of sequences in the set. if there are not enough sequences the `kmermatcher` module will fail. This should be fixed at the current version a357368b3e336a6a42772fb085fbc141df50b8ac. Could you please checkout the most recent version and rerun it?

**Comment by gmteunisse** [comment]
Updating seems to have fixed the problem. Thank you for your quick and effective response!
